src/sinks/postgres/config.rs

```diff
@@ -0,0 +1,144 @@
+use futures::FutureExt;
+use tower::ServiceBuilder;
+use vector_lib::{
+    config::AcknowledgementsConfig,
+    configurable::{component::GenerateConfig, configurable_component},
+    sink::VectorSink,
+};
+
+use super::{
+    service::{PostgresRetryLogic, PostgresService},
+    sink::PostgresSink,
+};
+use sqlx::{postgres::PgPoolOptions, Pool, Postgres};
+
+use crate::{
+    config::{Input, SinkConfig, SinkContext},
+    sinks::{
+        util::{
+            BatchConfig, RealtimeSizeBasedDefaultBatchSettings, ServiceBuilderExt,
+            TowerRequestConfig, UriSerde,
+        },
+        Healthcheck,
+    },
+};
+
+const fn default_pool_size() -> u32 {
+    5
+}
+
+/// Configuration for the `postgres` sink.
+#[configurable_component(sink("postgres", "Deliver log data to a PostgreSQL database."))]
+#[derive(Clone, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct PostgresConfig {
+    // TODO: if I used UriSerde instead of String, I couldn't get a url string to use
+    // in the connection pool, as the password would be redacted with UriSerde::to_string
+    /// The connection string for the PostgreSQL server. It can contain the username and password.
+    pub endpoint: String,
+
+    /// The table that data is inserted into.
+    pub table: String,
```

Comment:
Should we make the table templatable? Like the `clickhouse`  sink. That would complicate the code a little bit (with `KeyPartitioner` and so. If yes, I would like some guidance about it if possible

Reply:
It is a nice feature but not a must-have, we can do this incrementally. Once we finalized the rest of the comments we can come back to this if you are motivated to add this feature.